### ISSUE TITLE
Fragile servers (simulnet) tests

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#1175]
Occasionally we see something like the following in `grin_servers` tests.
This seems to be particulalry brittle during the `simulate_fast_sync` test.
  
```
Jun 16 11:31:36.880 WARN sync: no peers available, disabling sync
Jun 16 11:31:36.881 INFO Starting HTTP API server at 127.0.0.1:22001.
Jun 16 11:31:36.881 WARN Error accepting peer 127.0.0.1:65430: Connection(Custom { kind: WouldBlock, error: StringError("read_exact") })
Jun 16 11:31:36.881 DEBG connect_and_req: 127.0.0.1:12000 is Defunct; Serialization(UnexpectedData { expected: [30], received: [0] })
Jun 16 11:31:36.882 DEBG monitor_peers: 0 connected (0 most_work). all 0 = 0 healthy + 0 banned + 0 defunct
Jun 16 11:31:36.882 DEBG monitor_peers: no dandelion relay updating
Jun 16 11:31:36.882 DEBG Could not update dandelion relay
Jun 16 11:31:36.882 INFO Starting dandelion monitor: 127.0.0.1:22001
Jun 16 11:31:36.882 DEBG Started Dandelion transaction monitor.
Jun 16 11:31:36.883 WARN Grin server started.
```

The two test nodes have failed to peer successfully for some reason.
We see an error during the initial peering attempt -
```
Jun 16 11:31:36.881 WARN Error accepting peer 127.0.0.1:65430: Connection(Custom { kind: WouldBlock, error: StringError("read_exact") })
```

Followed by- 
```
Jun 16 11:31:36.881 DEBG connect_and_req: 127.0.0.1:12000 is Defunct; Serialization(UnexpectedData { expected: [30], received: [0] })
```

At that point we have two disconnected nodes and the test never finishes (it eventually times out on travis).

**Comment by antiochp** [comment]
When we do see the `simulate_fast_sync` test succeed it does not succeed for the right reasons...

```
Jun 16 11:37:40.599 DEBG Received block d3c330a0 at 29 from 127.0.0.1:12000, going to process.
Jun 16 11:37:40.600 DEBG pipe: process_block d3c330a0 at 29 with 0 inputs, 1 outputs, 1 kernels
Jun 16 11:37:40.600 DEBG process_block: orphan: d3c330a0, # orphans 1
Jun 16 11:37:40.600 DEBG adapter: process_block: received an orphan block, checking the parent: 45528ed4
Jun 16 11:37:40.600 DEBG Requesting block 45528ed4 from peer 127.0.0.1:12000.
Jun 16 11:37:40.602 DEBG Received block 45528ed4 at 28 from 127.0.0.1:12000, going to process.
Jun 16 11:37:40.602 DEBG pipe: process_block 45528ed4 at 28 with 0 inputs, 1 outputs, 1 kernels
Jun 16 11:37:40.602 DEBG process_block: orphan: 45528ed4, # orphans 2
Jun 16 11:37:40.602 DEBG adapter: process_block: received an orphan block, checking the parent: a284531a
Jun 16 11:37:40.602 DEBG Requesting block a284531a from peer 127.0.0.1:12000.
Jun 16 11:37:40.605 DEBG Received block a284531a at 27 from 127.0.0.1:12000, going to process.
Jun 16 11:37:40.605 DEBG pipe: process_block a284531a at 27 with 0 inputs, 1 outputs, 1 kernels
Jun 16 11:37:40.605 DEBG process_block: orphan: a284531a, # orphans 3
Jun 16 11:37:40.606 DEBG adapter: process_block: received an orphan block, checking the parent: f8259279
Jun 16 11:37:40.606 DEBG Requesting block f8259279 from peer 127.0.0.1:12000.
```

There is no "fast sync" occurring during the test run.
We see two nodes, neither of which are in `currently_syncing` mode.
The second node is receiving blocks via boradcast/gossip, identifying them as orphans and sequentially requesting the missing blocks.
This process eventually completes, the nodes are synchronized and the test "passes".

**Comment by antiochp** [comment]
Resolved by #1277.
